src/core/lint-hrefs-ids.js

```diff
@@ -0,0 +1,21 @@
+// Module core/lint-hrefs-ids
+// Warns about href's that do not have corresponding id's
+
+import { pub } from "core/pubsubhub";
+
+export const name = "core/lint-hrefs-ids";
+
+export function run(conf, doc, cb) {
+  console.log()
+  Array.from(doc.querySelectorAll('a[href]'))
+    .map(a => ({ a, href: a.getAttribute('href') }))
+    .filter(({ href }) => Boolean(href))
+    .filter(({ href }) => href.startsWith('#'))
+    .map(({ a, href }) => ({ a, href, id: href.slice(1) }))
+    .filter(({ id }) => Boolean(id))
```

Comment:
But falsy `getElementById()` means that the link is "offending". This `filter` currently only handles the case of an empty string, this could be done above with something like `href.startsWith('#') && href.length > 1` or `/^#.+/.test(href)` but I don't thing it's cleaner that way.